tests/testthat/helper.R

```diff
@@ -127,9 +132,29 @@ dbg <- function(..., .display = utils::str) {
     cat(loc, "\n")
   } else {
     cat(sprintf("(from call: %s (srcfile missing))\n", trimws(
-      deparse1(sys.call(-2), width.cutoff = 60)
+      deparse1(sys.call(-2) %error% sys.call(-1), width.cutoff = 60)
     )))
   }
 
   invisible(out)
 }
+
+`%error%` <- function(x, y) tryCatch(x, error = function(e) y)
+
+# prevent new_class() from creating `S7::` prefixed S3 class names in tests.
+{
+  body(topNamespaceName) <- bquote({
+    if (testthat::testing_package() == "S7") {
```

Comment:
I meant just do it in the typical definition of `topNamespaceName` so we don't need to do any messing around in here. (Which is much harder to discover if something goes wrong)

Reply:
Ideally, I think we want "test drive" the function as it might exist in base, i.e., without accommodations for the S7 unit-tests edge-case.

Reply:
Hmmmm, I just generally worry that this sort of magic eventually will cause a lot of head scratching when investigating a test failure (which is the time when you least want any surprises/magic). How much work would it be to explicitly update the tests?

Reply:
`new_class(` appears 234 times in the tests, each call would have to be updated. Happy to do it if that's the best way. 
```r
$ rg -F 'new_class(' tests | wc -l
234
```

Could also do something like this in `tests/helpers.R`:
```r
formals(new_class)['package'] <- list(NULL)
```

Reply:
Yeah, but it only causes ~20 non-snapshot test failures, so I think you could just update those?